doc/external_variables.rst

```diff
@@ -0,0 +1,75 @@
+==================
+External variables
+==================
+
+It is possible to pass external variables to Anabot, using either
+kernel command line or Beaker task parameters. The main use
+of externally defined variables is for enhancing recipes, where
+a variable placeholder can be substituted by Anabot's preprocessor,
+thus providing a greater flexibility in comparison with just static
+recipes.
+
+.. note::
+    Even though the main reason for introducing external variables
+    is their use in recipes, it is technically possible to use
+    those variables generally in Anabot's code, since those variables
+    are internally handled by |get_variable|_ and |set_variable|_.
+
+    .. |get_variable| replace:: ``get_variable()``
+    .. _get_variable: https://github.com/rhinstaller/anabot/blob/main/anabot/variables.py
+    .. |set_variable| replace:: ``set_variable()``
+    .. _set_variable: https://github.com/rhinstaller/anabot/blob/main/anabot/variables.py
+
+.. warning::
+    When defining external variables, chooose their names wisely.
+    As mentioned in the note above, they occupy the shared internal
+    variables space, which will lead to a clash when another variable with
+    the same name is used elsewhere in Anabot's code, leading to
+    undefined behaviour.
+
+    A list of already existing variables in Anabot code using this
+    mechanism can be obtained for instance by running the following command:
+    in Anabot repository:
+    ``git grep -Pho "(get|set)_variable\([\"']\K[^'\"]+" | sort -u``
+
+.. warning::
+    If a variable is defined both on kernel command line and in Beaker
+    task parameter, the Beaker one takes precedence. However, it is best
+    to avoid defining the same variable using both mechanisms at the same time.
```

Comment:
Why have Beaker parameters higher priority than the ones from the kernel command line? I would expect it the other way.
(Although I noticed this during the initial review and wrote a comment, it was not sent for review for some reason, I'm sorry Jirko)

Reply:
I considered the Beaker parameters to be more 'user visible' in our testing workflow (i. e. the Beaker parameters are the last place where user can change Anabot's behaviour in common test automation chain). However, I don't have a strong opinion about what order is better, so I see no problem with reversing the prioritization.

BTW, why would you expect the opposite?

Reply:
If I submit a Beaker job and I realize I want to change a variable value (for example because of debugging), manually modifying the kernel cmdline is a way to do it when the job is already running.

It is also common for tools that options specified on the command line have higher priorities than options from config files or environment variables. In this case I see Beaker parameters at the same level as environment variables.

Reply:
Thank you for your insight. I updated the code accordingly to give a higher priority to command line variables.